### PR TITLE
feat: resources are updated automatically once a new version is released

### LINKED ISSUE #216: Intelligent Installer Update Checks
### User Story

**Title**: Intelligent Installer Update Checks  
**As a** Captain installer  
**I want** to check for changes in essential resources rather than just version numbers  
**So that** users are only prompted to reinstall if there have been updates to the essential resources necessary for Captain to function

### Acceptance Criteria

- [x] The installer logic is updated to check a manifest of the essential resources, not just the version number in `package.json`.
- [x] If the version of the currently installed resources matches the version of the resources in the manifest of the latest release, the installer does not prompt for reinstallation.
- [x] If there is a mismatch in versions, indicating a change in essential resources, the installer prompts the user to download and install the updated resources.
- [x] The installer provides clear messaging to the user about why a reinstallation is necessary if prompted.

### PR DESCRIPTION
**Pull request by TimPietrusky**
## Motivation

* Provides an automated way to handle version updates of Captain
* When the resources (like embedded-python) are updated in the manifest, it will show the installer
* If no resource changed with an update of Captain, Captain will just open

## Issues closed

closes #216

### CODE REVIEW

File: src/client/ions/hooks/color-scheme.ts [hunk 1]
```diff
@@ -12,9 +13,10 @@ export function useSsrColorScheme() {
 		},
 		[setMode]
 	);
-
 	useEffect(() => {
 		setMode_(mode ?? "system");
+
+		window.ipc.send(USER_THEME_KEY, mode);
```
Comment (pixelass) [inline src/client/ions/hooks/color-scheme.ts]:
```suggestion
		window.ipc.send(USER_THEME_KEY, mode ?? "system");
```

File: src/client/pages/[locale]/installer/02-update.tsx [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+import type { ReactNode } from "react";
+import { useCallback, useEffect, useState } from "react";
+
+import { buildKey } from "#/build-key";
+import { ID } from "#/enums";
+import { makeStaticProperties } from "@/ions/i18n/get-static";
+import { Illustration } from "@/organisms/illustration";
+import { QuoteLoop } from "@/organisms/quote-loop";
+
+function useInstallProgress() {
```
Comment (pixelass) [inline src/client/pages/[locale]/installer/02-update.tsx]: isn't this the same as in `02?`

File: src/client/pages/[locale]/installer/02-update.tsx [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+			unsubscribeCancelled();
+			unsubscribeCompleted();
+			unsubscribeUnpacking();
+			unsubscribeFailed();
+		};
+	}, []);
+	return { status, progress, name, size, error, reset };
+}
+
+function InstallStep({
```
Comment (pixelass) [inline src/client/pages/[locale]/installer/02-update.tsx]: isn't this the same as in `02?`

**Review by pixelass — CHANGES_REQUESTED** [review]

File: src/client/pages/[locale]/installer/02-update.tsx (hunk 2 above)
Comment (TimPietrusky) [reply src/client/pages/[locale]/installer/02-update.tsx]: Yes, should I put it into `hooks`?

File: src/client/pages/[locale]/installer/02-update.tsx (hunk 3 above)
Comment (TimPietrusky) [reply src/client/pages/[locale]/installer/02-update.tsx]: Yes, should I put it into `molecules`?

Comment (pixelass) [reply src/client/pages/[locale]/installer/02-update.tsx]: organisms

**Review by pixelass — APPROVED** [review]

**Pull request merged**